cmd/kube-proxy/app/server.go

```diff
@@ -399,6 +399,20 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 	proxyMode := getProxyMode(string(config.Mode), iptInterface, iptables.LinuxKernelCompatTester{})
 	if proxyMode == proxyModeIPTables {
 		glog.V(0).Info("Using iptables Proxier.")
+		var nodeIP net.IP
+		if config.BindAddress == "0.0.0.0" || config.BindAddress == "" {
+			nodeIP = getNodeIP(client, hostname)
+		} else {
+			nodeIP = net.ParseIP(config.BindAddress)
+			if nodeIP == nil {
+				return nil, fmt.Errorf("bind-address %s must be valid ip", config.BindAddress)
+			}
+			if local, err := isLocalIP(nodeIP.String()); err != nil {
```

Comment:
Do we need to bother checking that here?  If it is not, it should fail with a good-enough error